src/main/scala/io/iohk/ethereum/forkid/ForkId.scala

```diff
@@ -0,0 +1,69 @@
+package io.iohk.ethereum.forkid
+
+import java.util.zip.CRC32
+import java.nio.ByteBuffer
+
+import akka.util.ByteString
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.utils.BigIntExtensionMethods._
+import io.iohk.ethereum.utils.ByteUtils._
+import io.iohk.ethereum.rlp._
+
+import RLPImplicitConversions._
+
+case class ForkId(hash: BigInt, next: Option[BigInt])
+
+object ForkId {
+
+  def create(genesisHash: ByteString, config: BlockchainConfig)(head: BigInt): ForkId = {
+    val crc = new CRC32()
+    crc.update(genesisHash.asByteBuffer)
+    val next = gatherForks(config).find { fork =>
+      if (fork <= head) {
+        crc.update(bigIntToBytes(fork, 8))
+      }
+      fork > head
+    }
+    new ForkId(crc.getValue(), next)
+  }
+
+  val noFork = BigInt("1000000000000000000")
+
+  def gatherForks(config: BlockchainConfig): List[BigInt] =
+    (config.daoForkConfig.map(_.forkBlockNumber).toList ++
+      config.forkBlockNumbers.productIterator.toList.flatMap {
```

Comment:
this assumes the order of fields in config case classes, right?

Reply:
The list is being sorted at the very end

Reply:
I would put that part with the `productIterator` inside `ForkBlockNumbers` directly. Even though it seem reasonable that every `BigInt` there will be a fork block number we never really know. If we put this function inside `ForkBlockNumbers` we make this assumption more explicit for devs modifying the case class.

If we want to avoid any logic in the config part, we could add a comment in ForkBlockNumbers which explain that every BigInt there must be a fork block number. 

What do you think ?